src/+groundwater/calculateGroundwaterRecharge.m

```diff
@@ -0,0 +1,131 @@
+
+                                %%%%%%%%%% Important note %%%%%%%%%%
+        The index order for the soil layers in STEMMUS is from bottom to top (NN:1), where NN is the number of STEMMUS soil layers,
+        which is the opposite of MODFLOW (top to bottom). So, when converting information between STEMMUS and MODFLOW, indices need to be flipped
+
+                                %%%%%%%%%% Variables definitions %%%%%%%%%%
+        Equations of the water balance are implemented in this function (Equations 8-13 of HYDRUS-MODFLOW paper) as follows:
+        recharge = recharge_init + (SY - sy) * DeltZ, where:
+
+        gwfluxes                structure that includes the outputs (groundwater recharge and its individual components)
+        recharge                groundwater recharge, which is the upper boundary flux of the top layer of the phreatic aquifer (after the correction of the specific yield)
+        recharge_init           upper boundary flux into the moving balancing domain (before the correction of the specific yield)
+        SS                      specific storage of MODFLOW aquifers
+        SY                      large-scale specific yield of the phreatic aquifer
+        sy                      small-scale specific yield (dynamically changing water yield) caused by fluctuation of the water table
+        depToGWT_strt           soil layer thickness from the top of the soil up to the phreatic surface at the start of the current time step
+        indxGWLay_strt          index of the soil layer that includes the phreatic surface at the start of the current time step
+        depToGWT_end            soil layer thickness from the top of the soil up to the phreatic surface at the end of the current time step
+        indxGWLay_end           index of the soil layer that includes the phreatic surface at the end of the current time step
+        Theta_L                 soil moisture at the start of the current time step (bottom to top)
+        Theta_LL                soil moisture at the end of the current time step (bottom to top)
+        STheta_L                soil moisture at the start of the current time step (top to bottom)
+        STheta_LL               soil moisture at the end of the current time step (top to bottom)
+        soilThick               cumulative soil layers thickness (from top to bottom)
+        indxAqLay               index of the MODFLOW aquifer that corresponds to each STEMMUS soil layer
+        aqLayers                elevation of top surface level and all bottom levels of aquifer layers, received from MODFLOW through BMI
+    %}
+
+    % Load model settings
+    ModelSettings = io.getModelSettings();
+    NN = ModelSettings.NN; % Number of nodes
+    NL = ModelSettings.NL; % Number of layers
+
+    % Start Recharge calculations
+    if GroundwaterSettings.GroundwaterCoupling == 1 % Groundwater coupling is enabled
```

Comment:
This **if statement** is not needed. This script is used for groundwater coupling and this function is used in the `STEMMUS_SCOPE.m` script under `if GroundwaterSettings.GroundwaterCoupling == 1`. Therefore, you can remove the if statement here. Please also remove the `end` and fix the indentation.

Reply:
> This **if statement** is not needed. This script is used for groundwater coupling and this function is used in the `STEMMUS_SCOPE.m` script under `if GroundwaterSettings.GroundwaterCoupling == 1`. Therefore, you can remove the if statement here. Please also remove the `end` and fix the indentation.

> 
Done